tools/sdk-generation-pipeline/.github/workflows/ci.yml

```diff
@@ -6,7 +6,7 @@ on:
     branches: [ "main" ]
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
```

Comment:
Note this is not a 1ES-hosted image, instead it is a [GitHub image](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#choosing-github-hosted-runners).

Reply:
@dw511214992 Do we use this github action anywhere? If so we should try to remain pinned to a version instead of floating via latest ad things tend to break when the move that reference.

@konrad-jamrozik this is github actions which currently doesn't support our 1ES agents.

Reply:
@weshaggard thanks, that's what I thought. Just to clarify: I believe the change should work, as it uses the [GitHub-hosted Ubuntu 22.04](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#choosing-github-hosted-runners), not 1ES agent.

Reply:
Yes the image update should be fine but we just cannot use our 1ES pool.

Reply:
@dw511214992 kindly asking for input on this :)

Reply:
I created a follow-up issue. Please see this comment:
- https://github.com/Azure/azure-sdk-tools/pull/5004#issuecomment-1398536891